anaBenn/Homework_#2.js

```diff
@@ -0,0 +1,533 @@
+
+//b) Write a second version of your function which improves the output in two special 
+//cases:
+
+//A output string like "0 1/2" should be simplified to just "1/2";
+//A output string like "1 0/3" should be simplified to just "1".
+
+
+var functionString = function(numerator,denominator){
+  var remainder = numerator % denominator;
+  var wholeNumber = Math.floor(numerator/denominator);
+  if(wholeNumber ===0){
+    return numerator + '/' + denominator;
+  }else if(numerator === 0){
+     return wholeNumber;
+  }else{
+     return wholeNumber + " " + remainder + '/' + denominator; 
+  }
+}; 
+console.log(functionString(8,3));
+
+
+//2) (Moderate, 10%)
+
+//Rewrite each block below as the simplest equivalent you can discover. Your equivalent should 
+//produce the same final conditions as the original code given the same initial conditions. In each 
+//case, if there is an undeclared variable, assume it has been declared earlier and set to an unknown value.
+
+//a)
+
+//var n = 55.43
+var i;
+if ((n - Math.floor(n)) >= 0.5) {
+  i = Math.ceil(n);
+} else {
+  i = Math.floor(n);
+}
+//console.log(i);
+//55
+
+
+//var n = 55.43;
+var i = Math.round(n);
+//console.log(i);
+//55
+
+
+
+b)
+
+var y, xIsFalse = (x? false : true);
+if (xIsFalse)
+  y = false;
+else
+  y = x;
+
+//
+
+var y = x || false
+
+//var y = x or var y = false.  || is the if else statement 
+
+
+c)
+
+//var i = 3;
+for (var count = 15 - i ; count < 15 ; count=count+1) {
+  i = i-1;
+  console.log(i+1);
+}
+//3
+//2
+//1
+
+//var i = 3;
+for ( ; i>0; i--) {
+    console.log(i);
+}
+//3
+//2
+//1
+
+d)
+
+//var a;
+//var b;
+var x;
+if (a) {
+  if (b) {
+    x = 0;
+  } else {
+    x = 1;
+  }
+} else {
+  if (b) {
+    x = 1;
+  } else {
+    x = 2;
+  }
+}
+//console.log(x);
+//2
+
+//var a;
+//var b;
+//var x;
+
+if (a && b) {
+   console.log (x = 0);
+}else if (a || b) {
+  console.log (x = 1);
+}else {
+  console.log (x = 2);
+} //2
+
+
+//3) (Moderate, 25%)
+
+//a) Write a function letterTriangle(n) which returns a single string as follows:
+
+//the function receives a parameter n, an integer from 1 to 26;
+//the output string will have n lines, each ending in a newline ('\n') character;
+//the first line of the string is always 'A';each line will be a sequence of letters, 
+//going backward from some starting letter to 'A';each line starts one letter after the 
+//previous line's start.As an example, the result of letterTriangle(4) will look like this 
+//when printed:
+
+
+var alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p",
+"q", "r", "s", "t", "u", "v", "w", "x", "y", "z"];
+
+var newAlphabet = [];
+
+var display = function(n){
+ newAlphabet.unshift(alphabet[n]);
+ console.log(newAlphabet);
+}
+//unshift will always do the new value first...that's why it's not pritning as a ab abc abcd etc.  
+//If it was .push instead of unshift, it would print that way.  
+
+for (i=0; i< alphabet.length; i++){
+ display(i);  
+}
```

Comment:
This is SO CLOSE! It's printing what you want into the console, but, if we run newAlphabet[4], we would get "v", or if we run newAlphabet[3], we would get "w". We want it to print a string of the letters going backwards. So, we would want to potentially use another helper function to do so. For reference, here is one of Dan's solutions that looks somewhat similar to yours: 

function letterTriangle(n) {
	var letterStr = 'ZYXWVUTSRQPONMLKJIHGFEDCBA\n',
		lines = '';
	for (var i=0; i<n; ++i) {
		lines += letters.slice(-i-2);
	}
	return lines;
}

 Overall, nice work on this one, though :)